src/import-data/ImportData.ts

```diff
@@ -65,16 +65,29 @@ export const ImportData = (props: ImportDataProps): ReactNode => {
   const importPFB = async (importRequest: PFBImportRequest, workspace: WorkspaceInfo) => {
     if (workspace.cloudPlatform === 'Azure') {
       const wdsUrl = await Ajax().Apps.listAppsV2(workspace.workspaceId).then(resolveWdsUrl);
-      await Ajax().WorkspaceData.startImportJob(wdsUrl, workspace.workspaceId, {
+      const { namespace, name, cloudPlatform } = workspace;
+      const { jobId } = await Ajax().WorkspaceData.startImportJob(wdsUrl, workspace.workspaceId, {
         url: importRequest.url.toString(),
         type: 'PFB',
       });
+      // TODO a better way to make this conform to expected shape?
+      asyncImportJobStore.update(
+        Utils.append({
+          targetWorkspace: { namespace, name, workspace: { cloudPlatform, workspaceId: workspace.workspaceId } },
+          jobId,
+          proxyUrl: wdsUrl,
+        })
+      );
+      notifyDataImportProgress(jobId);
```

Comment:
Seems like you might be able to pull `notifyDataImportProgress` here and on line 91 out to outside the conditional.